addons/mrp/models/stock_move.py

```diff
@@ -97,10 +97,24 @@ def _get_aggregated_product_quantities(self, **kwargs):
         """
         aggregated_move_lines = super()._get_aggregated_product_quantities(**kwargs)
         kit_name = kwargs.get('kit_name')
-        if kit_name:
-            for aggregated_move_line in aggregated_move_lines:
-                if aggregated_move_lines[aggregated_move_line]['description'] == kit_name:
+
+        to_be_removed = []
+        for aggregated_move_line in aggregated_move_lines:
+            bom_lines = aggregated_move_lines[aggregated_move_line]['product'].bom_line_ids
+            is_phantom = bom_lines[:1].bom_id.type == 'phantom'
+            description = aggregated_move_lines[aggregated_move_line]['description']
+
+            if kit_name:
+                if not is_phantom:
```

Comment:
(Sorry, I forgot to send this comment before to finish my review :/ )

There is an issue when the same component is used by different kits and at least two of those kits are in the delivery.
For example, take the steps to reproduce the issue but add a second kit (K2) using C1 and C3 as component, and add this kit in the sale quotation too.
By doing that, C1 will be two times in both K1 and K2 sections in the delivery print.

The cause of this issue are:
1. we take all the product's BOM lines and check the BOM of the first one instead of taking only the BOM line linked to the move;
2. we only check if the BOM is a kit instead of checking it's the current one (regarding the `kit_name`).

A way to fix that is to add the move's BOM inside the aggregated values and then use it here in this override. Something like that for example:
```suggestion
            bom = aggregated_move_lines[aggregated_move_line]['bom_id']
            is_phantom = bom.type == 'phantom'
            description = aggregated_move_lines[aggregated_move_line]['description']

            if kit_name:
                if not is_phantom or bom.display_name != kit_name:
```
That implies the aggregated values are not defined directly in `_get_aggregated_product_quantities` but in a new method to be able to override it in `mrp` to add the move's BOM too.